### PR TITLE
Nanites rework: a new no-brainer

### PR DESCRIPTION
**Pull request by Jordanstolethepreciousslam**
## About The Pull Request

Nanites now heal only up to 8 health each time user receive damage(not instantly)
Nanites bottles removed from NanoMedPlus
Nanites replication rate reduced by 33%
Nanites replication rate above 30u reduced by 50%

In a nutshell - nanites will no longer heal people out of combat and will restore slower, effectively making them a reactive healing tool rather than passive. 
This is a draft as im not sure whether nanites need higher blood cost or just moved back to researcher. For now it seems to be a huge power creep.

## Why It's Good For The Game

Nanites are somewhat gimmicky chem that let players pick harder fight at the cost of having less fights overall. However, without (nano)blood they are difficult to manage as they tend to waste 100+ units of blood on wounds you would rather heal with pills, UNLESS you use vali to heal damage before it eats your blood. 
And then there are nanite bottles, defeating the purpose of chem that eats your blood to heal you. It doesn't make sense and it encourages people that have no respect for their time to make nanite pills and cheese the whole blood management thing.

Theory time
Every 1u of nanites heal 4 health(2/0.5)
Every 1u of nanites consumes 13u of blood(2/0.15)
Every tick you can regain up to 1.5u of blood by mixing nutriments(0.4), iron(0.8) and QC(0.2)
Therefore it takes about 9 ticks to compensate bloodloss from 1u of nanites(13/1.5)
Premed unga with nanites spends about 1u of nanites per 10 damage(half of that damage is healed by premed, another by nanites)
Therefore for every 10 damage taken unga have to stay out of combat for 9 ticks to regain blood. If i remember correctly 1 tick roughly equals to 1.7 seconds. So, 15 seconds for every 10 damage(15*1.7)
For every 40 damage it's a minute. 80 is two. 120 is three
Yes, waiting three minutes after being beaten to a pulp sounds fair, but compare to premed only(120/2.3 = 52) that takes about  1.5 minutes(52*1.7)

Caught a pair of stray bumps? half a minute of recovery. Got into acid cloud? another 30 seconds. caught fire? a full minute. Friendly fire? another minute. Oh, there is also shrapnel? Bleeding? Consider yourself gimped 
Don't forget that you just can't use nanites without DD though, as you will become obese too fast. 
So to nanites for it's highest potential, you need bktt, nutriment/iron pills, DD, QC and Inaprovaline. Don't even try playing without premed though you'll be fucked up. 
At this point even premedding with meraderm sounds reasonable.

To simply put, my problem with nanites is that they are not really viable for frontliners(except for vali users). It's a free health for engineers, medics, snipers, and civilians. everyone but those who actually take damage
## Changelog
:cl:
balance: Nanites now use healing stacks acquired by taking damage. Each time user is hurt his healing stacks are set to 4 and spent every tick nanites heal them. If stacks reach zero, healing stops. Each damage type consumes 1 healing stack per tick, for a total of 4 ticks of healing single type of damage, and 2 ticks of healing if both damage type present.
balance: Nanites replication amount per tick below 30u from 0.15 to 0.10(blood consumption is not adjusted so cost per nanite is increased)
balance: Nanites replication amount per tick below 35u from 0.10 to 0.05
delete: Nanites bottles removed from NanoMedPlus
/:cl:

### CODE REVIEW

**Comment by goose4429** [comment]
Just a reminder: you can heal normally with nanites. No penalties. Just eat that tricord pill.

Nanites require higher skill to use than "eat funny red pill", but you can manage.
For more info, see front-page guide: https://tgstation13.org/wiki/TGMC:Nanites_guide

Marine nerf so insta-approve.

**Comment by Jordanstolethepreciousslam** [comment]
Old nanites. premed out of combat unga gets 120 damage. heals fast but loses ~10u of nanites(~130u blood).

https://user-images.githubusercontent.com/78968085/173445046-1a07ebb6-2f3a-452f-a57c-54b4d1426f9c.mp4

**Comment by Jordanstolethepreciousslam** [comment]
There is a hidden synergy between nanites and vali, and i envy vali users so much i simply add that synergy to nanites powercreeping that shit straight to the moon

**Comment by Frinid1** [comment]
So nerfed nanites will be for masochists to punch themself every now and then to proc healing, that sounds stupid as hell

**Comment by goose4429** [comment]
> So nerfed nanites will be for masochists to punch themself every now and then to proc healing, that sounds stupid as hell

It would waste nanites, and they regen very slowly. So they _can_ do it, but doing it is not a great idea.

**Comment by Arctrooper8** [comment]
What's wrong with nanites as they are currently? Right now they're a high risk high reward chem that rewards good blood management.

**Comment by Jordanstolethepreciousslam** [comment]
> Right now they're a high risk high reward chem that rewards good blood management.
They do not reward you for taking risks no. You simply can't die of them unless you have no idea how they work. No, they reward you for not actually playing the damn game. Nanites gameplay is taking one fight and turning into backline bitch for next 5 minutes as you watch your blues short bar becomes long and green

**Comment by Frinid1** [comment]
So instead you wanna make them into "Fight for few minutes and turn in backline bitch for 10 minutes IF YOU SURVIVE" because you reduce nanites per blood gain keeping healing same
Your "nobrainer" will makes things worse in fight because you get punched often for prolonged time
If you get seriously punched in small instance (which is whole point of this rework) and don't administer medicine to prevent nanites healing you for no reason it's generally skill problem, which just make nanites useless, because YOU'LL NEED MEDICINE
### **TLDR** Fuck that rework - makes nanites useless thrash that will kill you even faster, just make that they wont eat blood lover 55%

**Comment by Autisem** [comment]
I feel that this sorta defeats the point of using nanites. id rather just have them heal slower and be more efficient on blood consumption(maby 2 versions? 1 riodedcombat and another more efficient one) nanites fill a neech for more supportive roles to fill a void of carrying no meds but dealing with minor chip damage. if your in real trouble and get stomped by xenos youd need the corpsman either way

**Comment by zzurge** [comment]
nanites were and still are a noob trap to me,too much effort for that shit

**Comment by TiviPlus** [comment]
draft with no changes for 11 days, closing as stale
